daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md

```diff
@@ -0,0 +1,96 @@
+---
+type: docs
+title: "Dapr Visual Studio Code extension overview"
+linkTitle: "Dapr extension"
+weight: 10000
+description:  "How to develop and run Dapr applications with the Dapr extension"
+---
+
+
+Dapr offers a *preview* [Dapr Visual Studio Code extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-dapr) for local development and debugging of your Dapr applications.
+
+<a href="vscode:extension/ms-azuretools.vscode-dapr" class="btn btn-primary" role="button">Open in VSCode</a>
+
+## Features
+
+### Scaffold Dapr tasks 
+
+* Helps scaffold VS Code task.json and launch.json configurations needed to debug your application within the Dapr environment.
+* Requires an already existing launch.json files to be found in the VS Code workspace.
+* Example NodeJS app:
+
+After scaffolding the launch.json should contain similar snippets:
+```json
+        {
+            "type": "pwa-node",
+            "request": "launch",
+            "name": "Launch Program with Dapr",
+            "skipFiles": [
+                "<node_internals>/**"
+            ],
+            "program": "${workspaceFolder}/app.js",
+            "preLaunchTask": "daprd-debug",
+            "postDebugTask": "daprd-down"
+        }
+```
+After scaffolding the task.json should contain similar snippets:
+
+```json
+		{
+			"appId": "nodeapp",
+			"appPort": 3500,
+			"label": "daprd-debug",
+			"type": "daprd"
+		}
+```
+
+### Scaffold Dapr components 
+* Generates the Dapr component assets needed for general Dapr applications.
+* Provides a Component folder in your VS Code workspace that contains pubsub, statestore and a zipkin yaml file.
+* Example pubsub file:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: pubsub
+spec:
+  type: pubsub.redis
+  metadata:
+  - name: redisHost
+    value: localhost:6379
+  - name: redisPassword
+    value: ""
+```
```

Comment:
```suggestion
### Scaffold Dapr components

When adding Dapr to your application, you may want to have a dedicated components directory, separate from the default components initialized as part of `dapr init`.

To create a dedicated components folder with the default `statestore`, `pubsub`, and `zipkin` components, use the `Dapr: Scaffold Dapr Components` [Command Palette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette) operation.

1. Open your application directory in Visual Studio Code
1. Open the Command Palette with `Ctrl+Shift+P`
1. Select `Dapr: Scaffold Dapr Components`
1. Run your application with `dapr run --components-path ./components -- ...`
```

Reply:
Same comment here on "How" to use the extension.